_assets/js/click-tracking.js

```diff
@@ -0,0 +1,43 @@
+// Import our constants from constandts.js
+import { CLICK_TRACKING_ENDPOINT, ACCESS_KEY, AFFILIATE, MODULE_CODE } from './utils/constants';
+
+export default function clickTracking() {
+  // Get all the anchor tags that are children of b tags on the page:
+  const allLinks = document.querySelectorAll('.search-result');
+
+  // Loop through all the links and add a click event listener to each one:
+  allLinks.forEach((link, index) => {
+    const url = link.href;
+    const query = location.search.split('&').find((param) => param.startsWith('query='));
+    const position = location.search.split('&').includes('offset=') ?
+    Number(
+        location.search
+          .split('&')
+          .find((param) => param.startsWith('offset='))
+          .replace('offset=', '')
+      ) +
+      index +
+      1 : index + 1;
+    // What we need to send to the click tracking endpoint:
+    // url clicked
+    // search query
+    // affiliate
+    // position
+    // module code
+    // access key
```

Comment:
Probably not, but do we need to wrap any of these in [encodeURI](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI)?

Reply:
Yes, we need to wrap the getSearchParam output in encodeURI because it return spaces between multi-word queries. It probably won't happen for the offset values, but I'll just do it anyway since it's returning a string.

Reply:
As for affiliate, module code and access key, those never change so we shouldn't need to escape anything there.

Reply:
And I'm gonna trust the urls in the link hrefs because they're provided by search.gov.